src/libraries/System.IO.Compression.ZipFile/src/System/IO/Compression/ZipFile.Create.cs

```diff
@@ -375,26 +376,34 @@ public static ZipArchive Open(string archiveFileName, ZipArchiveMode mode, Encod
                 if (includeBaseDirectory && di.Parent != null)
                     basePath = di.Parent.FullName;
 
-                foreach (FileSystemInfo file in di.EnumerateFileSystemInfos("*", SearchOption.AllDirectories))
```

Comment:
I see that `EnumerableFileSystemInfos` internally calls [`NormalizeInputs()`](https://github.com/dotnet/runtime/blob/53c42c933605cf86e5d9019457f839c954aeb5af/src/libraries/System.Private.CoreLib/src/System/IO/DirectoryInfo.cs#L192), which joins the paths and looks for any invalid characters: https://github.com/dotnet/runtime/blob/174e04336e4a073416cb0baebb2be72728b9d64b/src/libraries/System.Private.CoreLib/src/System/IO/Enumeration/FileSystemEnumerableFactory.cs#L27

And the new code calls `FileSystemEntry.ToFullPath()` which only joins paths but does not validate invalid characters: https://github.com/dotnet/runtime/blob/8d09271584895a22960984e65ea0a396706cf84d/src/libraries/System.Private.CoreLib/src/System/IO/Enumeration/FileSystemEntry.Unix.cs#L76

But I confirmed those validations aren't needed in the new code because they relate to matchType and the search pattern, which aren't being used here. So we're good.